scripts/apidoc/diff.ts

```diff
@@ -0,0 +1,96 @@
+import type { DocsApiDiffIndex } from './utils';
+import { nameDocsDiffIndexFile, pathDocsDiffIndexFile } from './utils';
+
+/**
+ * Loads the diff index from the given source url.
+ *
+ * @param url The url to load the diff index from.
+ */
+async function loadRemote(url: string): Promise<DocsApiDiffIndex> {
+  return fetch(url).then((res) => {
+    if (!res.ok) {
+      throw new Error(
+        `Failed to load remote diff index from ${url}: ${res.statusText}`
+      );
+    } else {
+      return res.json() as Promise<DocsApiDiffIndex>;
+    }
+  });
+}
+
+/**
+ * Loads the diff index from the given local path.
+ *
+ * @param path The path to load the diff index from. Should start with `file://` for cross platform compatibility.
+ */
+async function loadLocal(path: string): Promise<DocsApiDiffIndex> {
+  return import(path).then((imp) => imp.default as DocsApiDiffIndex);
+}
+
+/**
+ * Loads the diff index from the given source.
+ * If the source starts with `https://` it will be loaded from the remote url.
+ * Otherwise it will be loaded from the local path.
+ *
+ * @param source The source to load the diff index from.
+ */
+async function load(source: string) {
+  if (source.startsWith('https://')) {
+    return loadRemote(source);
+  } else {
+    return loadLocal(source);
+  }
+}
+
+/**
+ * Returns a set of all keys from the given entries.
+ *
+ * @param entries The entries to get the keys from.
+ */
+function allKeys(...entries: Record<string, unknown>[]): Set<string> {
+  return new Set(entries.map(Object.keys).flat());
+}
+
+/**
+ * Compares the remote and local diff index and returns the differences.
+ * The returned object contains the module names as keys and the method names as values.
+ * If the module name is `ADDED` or `REMOVED` it means that the module was added or removed in the local diff index.
+ *
+ * @param remoteDiffIndex The url to the remote diff index. Defaults to the next.fakerjs.dev diff index.
+ * @param localDiffIndex The path to the local diff index. Defaults to the local diff index.
+ */
+export async function diff(
+  remoteDiffIndex = `https://next.fakerjs.dev/${nameDocsDiffIndexFile}`,
+  localDiffIndex = `file://${pathDocsDiffIndexFile}`
+): Promise<Record<string, string[]>> {
+  const remote = await load(remoteDiffIndex);
+  const local = await load(localDiffIndex);
```

Comment:
I'll try again. Maybe this makes it more clear what I was trying to express [in my previous comment](https://github.com/faker-js/faker/pull/1755/files/b777cae0d065d3c79df903f4e41aafe53bde5f88#r1089773621).
```suggestion
  const remote = await loadRemote(remoteDiffIndex);
  const local = await loadLocal(localDiffIndex);
```

Reply:
I already got that, but the load method is there intentionally to allow `diff`ing differet targets e.g. compare two local branches with each other or the "released" version with the "about to be released" version.

I added my previously in-brain-only code to the diff script so I hope it gets more abvious why it is there:

````sh
pnpm run docs:diff [Target] [Source]
pnpm run docs:diff https://docs-diff-notification--serene-sprite-f3ef50.netlify.app/api-diff-index.json
````

Reply:
But `target` will always be treated as an URL and source will always be treated as a locale file, or not?

Reply:
What about:
````sh
pnpm run docs:diff https://v8.fakerjs.dev/api-diff-index.json https://next.fakerjs.dev/api-diff-index.json
````

@Shinigami92 (or anybody else): Should I remove the ability to pass both urls and local paths for both parameters?

Reply:
TBH... I dont really care 🤷

Reply:
I assume I will be the only one using this.
@xDivisionByZerox Should I remove it to keep it "simple"?

Reply:
TBH, due to the default values I assumed that `source` was always a locale file path. But checking the code I saw that nothing stops you from using locale or remote URLs for either argument.
Maybe that's why I got so confused over this entire thing - I simply assumed.

> @xDivisionByZerox Should I remove it to keep it "simple"?
Make it work first. Refactor later if required.
This looks fine to me now.